src/core/scope.c

```diff
@@ -428,8 +428,9 @@ static void scope_sigchld_event(Unit *u, pid_t pid, int code, int status) {
         unit_tidy_watch_pids(u, 0, 0);
         unit_watch_all_pids(u);
 
-        /* If the PID set is empty now, then let's finish this off */
-        if (set_isempty(u->pids))
+        /* If the PID set is empty now, then let's finish this off
+           (On unified we use proper notifications) */
```

Comment:
Well, I've just missed notification:
```sh
-bash-4.3# hostnamectl
   Static hostname: systemd-testsuite
         Icon name: computer-vm
           Chassis: vm
        Machine ID: 048108263d3b4063803e2253c74a359a
           Boot ID: 0b6522e4ad6349ec9bd0db8e16b094fe
    Virtualization: qemu
  Operating System: Fedora 23 (Cloud Edition)
       CPE OS Name: cpe:/o:fedoraproject:fedora:23
            Kernel: Linux 4.5.6-200.fc23.x86_64
      Architecture: x86-64

-bash-4.3# grep cgroup /proc/self/mounts
cgroup /sys/fs/cgroup cgroup2 rw,nosuid,nodev,noexec,relatime 0 0
```
This works fine:
```sh
-bash-4.3# systemd-run --scope sh -c 'cat </dev/zero >/dev/null' &
[1] 180
Running scope as unit: run-r9eeeada9366543e5859b06dc96d12238.scope

-bash-4.3# systemctl stop run-r9eeeada9366543e5859b06dc96d12238.scope
[1]+  Terminated              systemd-run --scope sh -c 'cat </dev/zero >/dev/null'
```

But
```sh
-bash-4.3# systemd-run --scope sh -c 'cat </dev/zero >/dev/null' &
[1] 197
Running scope as unit: run-r20e5ae0204e44f47bd05f1f28cf85cc7.scope

-bash-4.3# mount -o remount /sys/fs/cgroup/
mount: /sys/fs/cgroup not mounted or bad option

       In some cases useful info is found in syslog - try
       dmesg | tail or so.

-bash-4.3# systemctl stop run-r20e5ae0204e44f47bd05f1f28cf85cc7.scope
...HANGS...
```

I collected the output of `strace -v -s 500 -p1`:
* Good
```
07:08:07.796262 inotify_add_watch(6, "/sys/fs/cgroup/system.slice/run-rcc2b028c577a4d62b9e2b8d6a9c1f67b.scope/cgroup.events", IN_MODIFY) = 58
...
07:08:07.944077 read(6, ":\0\0\0\2\0\0\0\0\0\0\0\0\0\0\0", 272) = 16
07:08:07.945686 open("/sys/fs/cgroup/system.slice/run-rcc2b028c577a4d62b9e2b8d6a9c1f67b.scope/cgroup.events", O_RDONLY|O_CLOEXEC) = 19
07:08:07.946538 fstat(19, {st_dev=makedev(0, 20), st_ino=355, st_mode=S_IFREG|0444, st_nlink=1, st_uid=0, st_gid=0, st_blksize=4096, st_blocks=0, st_size=0, st_atime=2016/06/16-07:08:07.795491701, st_mtime=2016/06/16-07:08:07.795491701, st_ctime=2016/06/16-07:08:07.795491701}) = 0
07:08:07.953486 fstat(19, {st_dev=makedev(0, 20), st_ino=355, st_mode=S_IFREG|0444, st_nlink=1, st_uid=0, st_gid=0, st_blksize=4096, st_blocks=0, st_size=0, st_atime=2016/06/16-07:08:07.795491701, st_mtime=2016/06/16-07:08:07.795491701, st_ctime=2016/06/16-07:08:07.795491701}) = 0
07:08:07.961556 mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f724c58f000
07:08:07.962720 read(19, "populated 1\n", 4096) = 12
07:08:07.964119 read(19, "", 4096)      = 0
07:08:07.965399 read(19, "", 4096)      = 0
07:08:07.968053 close(19)               = 0
07:08:07.968647 munmap(0x7f724c58f000, 4096) = 0
07:08:07.969402 read(6, 0x7ffc0aaa3cc0, 272) = -1 EAGAIN (Resource temporarily unavailable)
...
07:08:17.342193 read(6, ":\0\0\0\2\0\0\0\0\0\0\0\0\0\0\0", 272) = 16
07:08:17.343964 open("/sys/fs/cgroup/system.slice/run-rcc2b028c577a4d62b9e2b8d6a9c1f67b.scope/cgroup.events", O_RDONLY|O_CLOEXEC) = 44
07:08:17.344819 fstat(44, {st_dev=makedev(0, 20), st_ino=355, st_mode=S_IFREG|0444, st_nlink=1, st_uid=0, st_gid=0, st_blksize=4096, st_blocks=0, st_size=0, st_atime=2016/06/16-07:08:07.795491701, st_mtime=2016/06/16-07:08:07.795491701, st_ctime=2016/06/16-07:08:07.795491701}) = 0
07:08:17.346155 fstat(44, {st_dev=makedev(0, 20), st_ino=355, st_mode=S_IFREG|0444, st_nlink=1, st_uid=0, st_gid=0, st_blksize=4096, st_blocks=0, st_size=0, st_atime=2016/06/16-07:08:07.795491701, st_mtime=2016/06/16-07:08:07.795491701, st_ctime=2016/06/16-07:08:07.795491701}) = 0
07:08:17.349293 mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f724c58f000
07:08:17.349883 read(44, "populated 0\n", 4096) = 12
07:08:17.352762 read(44, "", 4096)      = 0
07:08:17.353389 read(44, "", 4096)      = 0
07:08:17.353949 close(44)               = 0
07:08:17.358719 munmap(0x7f724c58f000, 4096) = 0
...
07:08:17.399578 rmdir("/sys/fs/cgroup/system.slice/run-rcc2b028c577a4d62b9e2b8d6a9c1f67b.scope") = 0
07:08:17.401951 inotify_rm_watch(6, 58) = 0
```
* Bad
```
07:10:48.218336 inotify_add_watch(6, "/sys/fs/cgroup/system.slice/run-r192bda8a2e5e49258e2b13d573681937.scope/cgroup.events", IN_MODIFY) = 62
...
07:10:48.345813 read(6, ">\0\0\0\2\0\0\0\0\0\0\0\0\0\0\0", 272) = 16
07:10:48.353392 open("/sys/fs/cgroup/system.slice/run-r192bda8a2e5e49258e2b13d573681937.scope/cgroup.events", O_RDONLY|O_CLOEXEC) = 19
07:10:48.354618 fstat(19, {st_dev=makedev(0, 20), st_ino=355, st_mode=S_IFREG|0444, st_nlink=1, st_uid=0, st_gid=0, st_blksize=4096, st_blocks=0, st_size=0, st_atime=2016/06/16-07:10:48.216497438, st_mtime=2016/06/16-07:10:48.216497438, st_ctime=2016/06/16-07:10:48.216497438}) = 0
07:10:48.355609 fstat(19, {st_dev=makedev(0, 20), st_ino=355, st_mode=S_IFREG|0444, st_nlink=1, st_uid=0, st_gid=0, st_blksize=4096, st_blocks=0, st_size=0, st_atime=2016/06/16-07:10:48.216497438, st_mtime=2016/06/16-07:10:48.216497438, st_ctime=2016/06/16-07:10:48.216497438}) = 0
07:10:48.364998 mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f724c58f000
07:10:48.394573 read(19, "populated 1\n", 4096) = 12
07:10:48.438814 read(19, "", 4096)      = 0
07:10:48.447005 read(19, "", 4096)      = 0
07:10:48.447985 close(19)               = 0
07:10:48.449225 munmap(0x7f724c58f000, 4096) = 0
07:10:48.458371 read(6, 0x7ffc0aaa3cc0, 272) = -1 EAGAIN (Resource temporarily unavailable)
...
07:12:39.670865 rmdir("/sys/fs/cgroup/system.slice/run-r192bda8a2e5e49258e2b13d573681937.scope") = 0
07:12:39.676508 inotify_rm_watch(6, 62) = 0
```

@htejun , is this a known issue?

I'll try to reproduce this on `4.7`

Notes:
```
qemu-system-x86_64 --version
QEMU emulator version 2.4.1 (qemu-2.4.1-10.fc23), Copyright (c) 2003-2008 Fabrice Bellard
```